tests/test_fuzzy_checker.py

```diff
@@ -1,7 +1,244 @@
+    ],
+)
+def test_fail_fuzzy_assert_proportion(
+    numerator: int, denominator: int, target_proportion: float, match: str
+) -> None:
+    with pytest.raises(AssertionError, match=match):
+        FuzzyChecker().fuzzy_assert_proportion(numerator, denominator, target_proportion)
+
+
+def test_small_sample_size_fuzzy_assert_proportion(caplog: LogCaptureFixture) -> None:
+    FuzzyChecker().fuzzy_assert_proportion(1, 10, 0.1)
+    assert "Sample size too small" in caplog.text
+
+
+def test_not_conclusive_fuzzy_assert_proportion(caplog: LogCaptureFixture) -> None:
+    # This test verifies we will pass, then be inconclusive, then fail.
+    # The numbers used in this test are arbitrary but are intended to be conservative
+    # estimates of the number of iterations needed to reach each state
+    # Creating an instance here allows us to cache some of the computation for the while loop
+    fuzzy_checker = FuzzyChecker()
+    i = 1_000
+    while True:
+        caplog.clear()
+        fuzzy_checker.fuzzy_assert_proportion(i, 10_000, 0.1)
+        if "is not conclusive" in caplog.text:
+            assert i > 1050
+            break
+        if i > 1_200:
+            raise RuntimeError("Test did not reach the expected warning")
+        i += 1
+
+    while True:
+        caplog.clear()
+        try:
+            fuzzy_checker.fuzzy_assert_proportion(i, 10_000, 0.1)
+            assert "is not conclusive" in caplog.text
+        except AssertionError as e:
+            assert "is significantly greater" in str(e)
+            assert i > 1_100
+            break
+        if i > 1_300:
+            raise RuntimeError("Test did not reach the expected warning")
+        i += 1
+
+
+@pytest.mark.parametrize("step", (-1, 1))
+def test__calculate_bayes_factor(step: int) -> None:
+    # This is the base case where our numerator / denominator = target_proportion
+    numerator = 10_000
+    denominator = 100_000
+    # Parametrize rv_discrete for no bug distribution
+    # I am keeping the defaults for the bug distribution to remain 0.5 for alpha and beta
+    bug_issue_distribution = scipy.stats.betabinom(a=0.5, b=0.5, n=denominator)
+    no_bug_issue_distribution = scipy.stats.binom(p=TARGET_PROPORTION, n=denominator)
+    bayes_factor = FuzzyChecker()._calculate_bayes_factor(
+        numerator, bug_issue_distribution, no_bug_issue_distribution
+    )
+    previous_bayes_factor = bayes_factor
+    assert isinstance(bayes_factor, float)
+    assert bayes_factor > 0
+    while numerator > 0 and numerator < 100_000:
+        numerator += step
+        bayes_factor = FuzzyChecker()._calculate_bayes_factor(
+            numerator, bug_issue_distribution, no_bug_issue_distribution
+        )
+        assert isinstance(bayes_factor, float)
+        assert bayes_factor > 0
+        # Break once we reach infinity
+        if bayes_factor == float("inf"):
+            break
+        assert bayes_factor - previous_bayes_factor >= float(np.finfo(float).min) * 1_000
```

Comment:
Perhaps a comment here like "check that Bayes factor is getting larger (except for small wiggles) as we move further from the target proportion"